src/response/error.rs

```diff
@@ -163,50 +181,116 @@ impl From<tonic::Status> for MomentoError {
 fn status_to_error(status: tonic::Status) -> MomentoError {
     log::debug!("translating raw status to error: {status:?}");
     match status.code() {
-        tonic::Code::InvalidArgument => MomentoError::BadRequest {
-            description: "invalid argument".into(),
-            source: Some(status.into()),
-        },
-        tonic::Code::Unimplemented => MomentoError::BadRequest {
-            description: "unimplemented".into(),
-            source: Some(status.into()),
-        },
-        tonic::Code::OutOfRange => MomentoError::BadRequest {
-            description: "out of range".into(),
-            source: Some(status.into()),
-        },
-        tonic::Code::FailedPrecondition => MomentoError::BadRequest {
-            description: "failed precondition".into(),
-            source: Some(status.into()),
-        },
-        tonic::Code::Cancelled => MomentoError::Cancelled {
-            description: "cancelled".into(),
-            source: status.into(),
-        },
-        tonic::Code::DeadlineExceeded => MomentoError::Timeout {
-            description: "timed out".into(),
-            source: status.into(),
-        },
-        tonic::Code::PermissionDenied => MomentoError::PermissionDenied {
-            description: "permission denied".into(),
-            source: status.into(),
-        },
-        tonic::Code::Unauthenticated => MomentoError::Unauthenticated {
-            description: "unauthenticated".into(),
-            source: status.into(),
-        },
-        tonic::Code::ResourceExhausted => MomentoError::LimitExceeded {
-            description: "resource exhausted".into(),
-            source: status.into(),
-        },
-        tonic::Code::NotFound => MomentoError::NotFound {
-            description: "not found".into(),
-            source: status.into(),
-        },
-        tonic::Code::AlreadyExists => MomentoError::AlreadyExists {
-            description: "already exists".into(),
-            source: status.into(),
-        },
+        tonic::Code::InvalidArgument => MomentoError::InvalidArgument(SdkError {
+            message: "Invalid argument passed to Momento client".into(),
+            error_code: MomentoErrorCode::InvalidArgumentError,
+            inner_exception: Some(status.clone().into()),
+            details: MomentoGrpcErrorDetails {
+                code: status.code(),
+                details: "invalid argument".into(),
+                metadata: Some(status.metadata().clone())
+            }.into()
+        }),
+        tonic::Code::Unimplemented => MomentoError::BadRequest(SdkError {
+            message: "The request was invalid; please contact us at support@momentohq.com".into(),
+            error_code: MomentoErrorCode::BadRequestError,
+            inner_exception: Some(status.clone().into()),
+            details: MomentoGrpcErrorDetails {
+                code: status.code(),
+                details: "unimplemented".into(),
```

Comment:
it looks to me like the `status` object has a function called `details`, I think you will want to call that rather than making up our own.

There's also a `message` fn. We should add a corresponding `message` field to ours and expose that.

Also I don't think that converting from `status` to these `MomentoGrpcErrorDetails` should differ at all from one case to the next, so I think you can define `impl From<tonic::Status> for MomentoGrpcErrorDetails` and then all of these will just be `details: Some(status.into())`, just like what you are doing for the inner_exception.